message/validation/signer_state.go

```diff
@@ -4,34 +4,20 @@ package validation
 
 import (
 	specqbft "github.com/ssvlabs/ssv-spec/qbft"
-
-	"github.com/attestantio/go-eth2-client/spec/phase0"
 )
 
 // SignerState represents the state of a signer, including its start time, slot, round,
 // message counts, proposal data, and the number of duties performed in the current epoch.
 type SignerState struct {
-	Slot               phase0.Slot
-	Round              specqbft.Round
-	MessageCounts      MessageCounts
-	ProposalData       []byte
-	EpochDuties        int
-	SeenDecidedLengths map[int]struct{}
+	Round         specqbft.Round
+	MessageCounts MessageCounts
+	ProposalData  []byte
+	SeenSigners   map[string]struct{}
 }
 
-// ResetSlot resets the state's slot, round, message counts, and proposal data to the given values.
-// It also updates the start time to the current time and increments the epoch duties count if it's a new epoch.
-func (s *SignerState) ResetSlot(slot phase0.Slot, round specqbft.Round, newEpoch bool) {
-	s.Slot = slot
-	s.Round = round
-	s.MessageCounts = MessageCounts{}
-	s.ProposalData = nil
-	if newEpoch {
-		s.EpochDuties = 1
-	} else {
-		s.EpochDuties++
-	}
-	s.SeenDecidedLengths = make(map[int]struct{})
+func (s *SignerState) Init() {
+	s.Round = specqbft.FirstRound
+	s.SeenSigners = make(map[string]struct{})
```

Comment:
Shouldn't this also restart `MessageCounts` and `ProposalData`?

Reply:
@y0sher looks like this is not used to reset SignerState, but only after creating it, so I just replaced this to NewSignerState and using this instead of struct literal.

With that said, ResetRound seems to reset everything except SeenSigners so I've fixed it also.